spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/ReplicaPartitioner.scala

```diff
@@ -0,0 +1,51 @@
+package com.datastax.spark.connector.rdd.partitioner
+
+import java.net.InetAddress
+
+import com.datastax.spark.connector.cql.CassandraConnector
+import org.apache.spark.{Partition, Partitioner}
+
+
+case class ReplicaPartition(index: Int, endpoints: Set[InetAddress]) extends EndpointPartition
+
+/**
+ * The replica partitioner will work on an RDD which is keyed on sets of InetAdresses representing Cassandra
+ * Hosts . It will group keys which share a common IP address into partitionsPerReplicaSet Partitions.
+ * @param partitionsPerReplicaSet The number of Spark Partitions to make Per Unique Endpoint
+ * @param connector
+ */
+class ReplicaPartitioner(partitionsPerReplicaSet: Int, connector: CassandraConnector) extends Partitioner {
+  /* TODO We Need JAVA-312 to get sets of replicas instead of single endpoints. Once we have that we'll be able to
+  build a map of Set[ip,ip,...] => Index before looking at our data and give the all options for the preferred location
+   for a partition*/
+  val hosts = connector.hosts.toVector
+  val numHosts = hosts.size
+  val partitionIndexes = (0 until partitionsPerReplicaSet * numHosts).grouped(partitionsPerReplicaSet).toList
+  val hostMap = hosts zip partitionIndexes toMap
```

Comment:
toMap is in postfix position here; postfix syntax is going to be deprecated (because it is errorprone). Preferred style:

```scala
(hosts zip partitionIndexes).toMap
```